### PR TITLE
Added function for Mentor OAuth

### PR DESCRIPTION
**Pull request by rakaar**
Mentor OAuth working fine. Tested using postman

Couldn't test from frontend as it was giving CORS error in spite of enabling it in `main.go`
Once check if it is the right way or not
if not please correct and push it

### CODE REVIEW

File: controllers/mentor.go [hunk 1]
```diff
@@ -1,14 +1,82 @@
...
+	mentorOAuth, _ := mentorOAuth1.(map[string]interface{})
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"message": "` + err.Error() + `"}`))
+		return
+	}
+
+	// using the code obtained from above to get AccessToken from Github
+	req, _ := json.Marshal(map[string]interface{}{
+		"client_id":     "74557dcb91016b10b54b",
```
Comment (grapheo12) [inline controllers/mentor.go:30]: Hardcoded values!! Move it to env file and load from there.

File: main.go [hunk 2]
```diff
@@ -36,9 +37,11 @@ func main() {
 
 	router := mux.NewRouter().StrictSlash(true)
 
+	router.HandleFunc("/mentor/oauth", controllers.MentorOAuth).Methods("POST")
 	router.HandleFunc("/mentor", controllers.MentorReg).Methods("POST")
 	router.HandleFunc("/project", controllers.ProjectReg).Methods("POST")
 	router.HandleFunc("/project/all", controllers.ProjectGet).Methods("GET")
 
-	log.Fatal(http.ListenAndServe(":"+port, router))
+	log.Fatal(http.ListenAndServe(":"+port, handlers.CORS()(router)))
```
Comment (grapheo12) [inline main.go:45]: Addressed the CORS issue in #12 . So, after that is merged. Resolve the merge conflict and then merge it.

File: main.go [hunk 3]
```diff
@@ -36,9 +37,11 @@ func main() {
 
 	router := mux.NewRouter().StrictSlash(true)
 
+	router.HandleFunc("/mentor/oauth", controllers.MentorOAuth).Methods("POST")
```
Comment (grapheo12) [inline main.go:40]: Let's divide the urls into subrouters and add only the subrouters in main.go. Subroutes can register urls in views. I will do that in another PR. It's fine for now.

Comment (rakaar) [reply main.go:40]: Acha cool !

**Pull request merged**